src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatibilityAnalyzer.cs

```diff
@@ -1590,16 +1623,11 @@ static Versions NormalizeAttribute(Versions attributes)
 
         private static bool TryAddValidAttribute([NotNullWhen(true)] ref SmallDictionary<string, Versions>? attributes, AttributeData attribute)
         {
-            if (!attribute.ConstructorArguments.IsEmpty &&
-                                attribute.ConstructorArguments[0] is { } argument &&
-                                argument.Kind == TypedConstantKind.Primitive &&
-                                argument.Type.SpecialType == SpecialType.System_String &&
-                                !argument.IsNull &&
-                                !argument.Value.Equals(string.Empty) &&
-                                TryParsePlatformNameAndVersion(argument.Value.ToString(), out string platformName, out Version? version))
+            if (HasNonEmptyStringArgument(attribute, out var argument) &&
+                TryParsePlatformNameAndVersion(argument, out var platformName, out var version))
```

Comment:
Is it worth creating a helper for `HasNonEmptyStringArgument(attribute, out var argument) && TryParsePlatsformNameAndVersion(argument, out var platformName, out var version)` since it's used in a couple places?

Reply:
I think the helper would need a long name, something like `TryParsePlatsformNameAndVersionForNonEmptyStringArgument`, isn't the existing condition is more readable/understandable?

Reply:
It could just be an overload of `TryParsePlatformNameAndVersion`.

```csharp
        private static bool TryParsePlatformNameAndVersion(AttributeData attribute, out string osPlatformName, [NotNullWhen(true)] out Version? version)
        {
            if (HasNonEmptyStringArgument(attribute, out string? argument))
            {
                return TryParsePlatformNameAndVersion(argument, out osPlatformName, out version);
            }

            version = null;
            osPlatformName = string.Empty;

            return false;
        }
```